### PR TITLE
fix load explanation methods for numpy based variables

### PR DESCRIPTION
**Pull request by imatiach-msft**
A user wanted to save and load explanations using save/load methods to json (without using explanationclient), however these were not deserializing the local importance values properly as a numpy array, leading to exceptions when viewed in widget.    This PR resolves this.  In the process of adding tests, I found a host of other issues:
1.) We weren't saving/loading some properties like eval_data, init_data and pred_y/pred_proba_y (fixed in this PR)
2.) The save/load explanation code can't handle sparse data, I've added a disabled test and created a github issue to track this (https://github.com/interpretml/interpret-community/issues/378)

### CODE REVIEW

File: test/test_serialize_explanation.py [hunk 1]
```diff
@@ -70,6 +82,7 @@ def _assert_explanation_equivalence(actual, expected):
         param = getattr(ExplainParams, paramkey)
         actual_value = getattr(actual, param, None)
         expected_value = getattr(expected, param, None)
+        print(param)
```
Comment (imatiach-msft) [inline test/test_serialize_explanation.py:85]: remove

Comment (imatiach-msft) [reply test/test_serialize_explanation.py]: done!

**Review by gregorybchris — APPROVED** [review]

File: python/interpret_community/common/constants.py [hunk 2]
```diff
@@ -95,6 +95,11 @@ def get_serializable(cls):
                 vars(cls).keys())) - set(['DATA_MAPPER', 'DATA_MAPPER_INTERNAL']))
 
 
+class PrivateExplainParams(object):
```
[the remark this answers is not in the document]
Comment (imatiach-msft) [reply python/interpret_community/common/constants.py:98]: my thoughts were to reuse this in azureml-interpret, I don't think there's harm in making the constants public?

[the remark this answers is not in the document]
Comment (gaugup) [reply python/interpret_community/common/constants.py:98]: Can this be a helper function that return '_' +  ExplainParams.EVAL_DATA. Seems like defining a lot of trivial public classes.

File: test/test_serialize_explanation.py [hunk 3]
```diff
@@ -88,14 +100,28 @@ def _assert_explanation_equivalence(actual, expected):
...
+        np.testing.assert_array_equal(actual._eval_data, expected._eval_data)
+
+
 # performs serialization and de-serialization for any explanation
 # tests to verify that the de-serialized result is equivalent to the original
 # exposed outside this module to allow any test involving an explanation to
 # incorporate serialization testing
-def verify_serialization(explanation, extra_path=None, exist_ok=False):
+def verify_serialization(explanation, extra_path=None, exist_ok=False, assert_types=False):
     path = 'brand/new/path'
```
Comment (gaugup) [inline test/test_serialize_explanation.py:120]: maybe assert_types can be renamed to assert_numpy_types.

**Review by gaugup — COMMENTED** [review]
Looks good. Left some comments.

File: test/test_serialize_explanation.py (hunk 3 above)
Comment (imatiach-msft) [reply test/test_serialize_explanation.py:120]: done!

File: python/interpret_community/common/constants.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (imatiach-msft) [reply python/interpret_community/common/constants.py]: done!

**Review by gaugup — APPROVED** [review]

**Pull request merged**